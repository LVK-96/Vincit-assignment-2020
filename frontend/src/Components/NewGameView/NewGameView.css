.new-game-container {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
}

.button-container {
  margin: 20px 20px 70px;
}

@media only screen and (max-width: 860px) {
  .new-game-container {
    justify-content: flex-end;
  }
}
