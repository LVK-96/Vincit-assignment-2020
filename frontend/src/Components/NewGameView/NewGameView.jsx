import React from 'react';
import utils from '../../utils';
import usersService from '../../Services/users';
import Button from '../Button';
import './NewGameView.css';

const NewGameView = ({ setPoints }) => {
  const handleNewGameClick = async () => {
    try {
      const user = await usersService.createUser();
      utils.setAndStoreToken(user.token);
      utils.setAndStoreId(user.id);
      setPoints(user.points);
    } catch (e) {
      window.alert('Failed to start game!');
    }
  };

  return (
    <>
      <div className="button-container">
        <Button className="game-button" text="Uusi peli" handleClick={handleNewGameClick} disabled={false} visible />
      </div>
    </>
  );
};

export default NewGameView;
