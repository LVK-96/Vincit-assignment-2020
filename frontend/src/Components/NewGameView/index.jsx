import NewGameView from './NewGameView';

export default NewGameView;
