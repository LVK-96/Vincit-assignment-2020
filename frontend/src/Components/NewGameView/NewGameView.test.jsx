import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NewGameView from './NewGameView';

afterEach(cleanup);

test('renders correctly', () => {
  const component = render(
    <NewGameView />
  );

  expect(component.container).toHaveTextContent(
    'Uusi peli'
  );
});

test('snapshot test', () => {
  const component = renderer.create(
    <NewGameView />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
