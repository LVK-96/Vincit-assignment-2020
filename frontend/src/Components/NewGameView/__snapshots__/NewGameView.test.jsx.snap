// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="button-container"
>
  <button
    className="game-button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Uusi peli
  </button>
</div>
`;
